### PR TITLE
Polygon fix - detonate file

### PR DESCRIPTION
**Pull request by karinafishman**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9192
relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-7850

## Description
- Added new task for filtering supported files by Polygon
- Added condition to the upload command to filter the file types (in case there are more then one file uploaded)
- Added image to the playbook

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 1]
```diff
@@ -371,23 +436,125 @@ tasks:
...
+      - "1"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: match
+          left:
+            value:
+              complex:
+                root: inputs.file_id
```
Comment (ssokolovich) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:518]: change it to the condition of File.Type + checking if the input.id is in the File.EntryID

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 2]
```diff
@@ -371,23 +436,125 @@ tasks:
...
+            iscontext: true
+          right:
+            value:
+              simple: .*(?:7Z|ACE|AR|BAT|ARJ|BAT|BZ2|CAB|CHM|CMD|COM|CPGZ|CPL|CSV|DAT|DOC|DOCM|DOCX|DOCT|DOTM|DOTX|EML|EXE|GZ|GZIP|HTA|THM|HTML|IQY|ISO|JAR|JS|JS|JSE|LNK|LZ|LZMA|IZO|IZH|MCL|MHTMSG|MSI|MSP|ODP|ODS|ODT|OTS|OTT|PDF|PIF|POTM|POTX|PPS|PPT|PPTM|PPTX|PS1|PUB|PY|PYC|RAR|REG|RTF|SCR|SETTINGSCONTENT-MS|STC|SVG|SXC|SXW|TAR|TAZ|TB2|TBZ|TBZ2|TGZ|TLZ|TXZ|TZO|TXT|URL|UUE|VBE|VBS|WSF|XAR|XLS|XLSB|XLSM|XLSX|XML|XZ|ZIP)\b
+          ignorecase: true
+        - operator: match
+          left:
+            value:
+              complex:
+                root: inputs.file_id
```
Comment (ssokolovich) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:529]: change it to the condition of File.Type + checking if the input.id is in the File.EntryID

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 3]
```diff
@@ -371,23 +436,125 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "20":
+    id: "20"
+    taskid: 2cec22a6-b48e-474b-84ee-4aec0f057084
+    type: regular
+    task:
+      id: 2cec22a6-b48e-474b-84ee-4aec0f057084
+      version: -1
+      name: 'Set report number to content '
```
Comment (ssokolovich) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:461]: Check if we need to use this task of `set` - if not pls update the PB + update the image

**Review by ssokolovich — APPROVED** [review]

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 4]
```diff
@@ -401,151 +540,153 @@ inputs:
       accessor: EntryID
   required: false
   description: File to detonate
```
Comment (ssokolovich) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:542]:
```suggestion
  description: The File EntryID which will be detonated.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 5]
```diff
@@ -400,152 +544,159 @@ inputs:
...
-  description: The actual score
+  description: The actual score.
   type: number
 - contextPath: IP.Address
-  description: IP address
+  description: IP address.
   type: string
 - contextPath: Domain.Name
-  description: The Domain name
+  description: The Domain name.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:604]:
```suggestion
  description: The domain name.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 6]
```diff
@@ -400,152 +544,159 @@ inputs:
...
-  description: Process name
+  description: Process name.
   type: string
 - contextPath: Process.PID
-  description: Process PID
+  description: Process PID.
   type: number
 - contextPath: Process.CommandLine
-  description: Process Command Line
+  description: Process Command Line.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:625]:
```suggestion
  description: Process command line.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 7]
```diff
@@ -400,152 +544,159 @@ inputs:
...
-  description: Process end time
+  description: Process end time.
   type: date
 - contextPath: Polygon.Analysis.ID
-  description: Analysis ID in THF
+  description: Analysis ID in THF.
   type: number
 - contextPath: Polygon.Analysis.Name
-  description: File Name
+  description: File Name.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:640]:
```suggestion
  description: File name.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 8]
```diff
@@ -400,152 +544,159 @@ inputs:
...
   type: string
 - contextPath: Polygon.Analysis.Size
-  description: File Size
+  description: File Size.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:643]:
```suggestion
  description: File size.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml [hunk 9]
```diff
@@ -400,152 +544,159 @@ inputs:
...
   type: unknown
 - contextPath: Polygon.Analysis.URL
-  description: The information about URL indicators
+  description: The information about URL indicators.
 - contextPath: Polygon.Analysis.IP
-  description: The information about IP indicators
+  description: The information about IP indicators.
 - contextPath: Polygon.Analysis.Domain
-  description: The information about Domain indicators
+  description: The information about Domain indicators.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon.yml:695]:
```suggestion
  description: The information about domain indicators.
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md [hunk 10]
```diff
@@ -1,96 +1,96 @@
...
 
-| **Name** | **Description** | **Default Value** | **Source** | **Required** |
-| --- | --- | --- | --- | --- |
-| File | The file object of the file to detonate. The file is taken from the context. | None | File | Optional |
-| Interval | The duration for executing the pooling (in minutes). | 1 | - | Optional |
-| Timeout | The duration after which to stop pooling and to resume the playbook (in minutes). | 60 | - | Optional |
-| Password | The password for the uploaded file. | - | - | Optional |
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| file_id | File to detonate | File.EntryID | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md:34]:
```suggestion
| file_id | File EntryID to detonate | File.EntryID | Optional |
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md [hunk 11]
```diff
@@ -1,96 +1,96 @@
...
+| File.SHA256 | The SHA256 hash of the file. | string |
+| File.Type | File type. | string |
+| File.Malicious.Vendor | The vendor that reported the file as malicious. | string |
+| File.Malicious.Description | A description explaining why the file was determined to be malicious. | string |
+| DBotScore.Indicator | The indicator that was tested. | string |
+| DBotScore.Type | The indicator type. | string |
+| DBotScore.Vendor | The vendor used to calculate the score. | string |
+| DBotScore.Score | The actual score. | number |
+| IP.Address | IP address. | string |
+| Domain.Name | The Domain name. | string |
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md:57]:
```suggestion
| Domain.Name | The domain name. | string |
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md [hunk 12]
```diff
@@ -1,96 +1,96 @@
...
+| Domain.DNS | A list of IP objects resolved by DNS. | string |
+| URL.Data | The URL. | string |
+| RegistryKey.Path | The path to the registry key. | string |
+| RegistryKey.Value | The value at the given RegistryKey. | string |
+| Process.Name | Process name. | string |
+| Process.PID | Process PID. | number |
+| Process.CommandLine | Process Command Line. | string |
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md:64]:
```suggestion
| Process.CommandLine | Process command line. | string |
```

File: Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md [hunk 13]
```diff
@@ -1,96 +1,96 @@
...
+| Process.Path | Process path. | string |
+| Process.StartTime | Process start time. | date |
+| Process.EndTime | Process end time. | date |
+| Polygon.Analysis.ID | Analysis ID in THF. | number |
+| Polygon.Analysis.Name | File Name. | string |
+| Polygon.Analysis.Size | File Size. | number |
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/Playbooks/playbook-Detonate_File_-_Group-IB_TDS_Polygon_README.md:70]:
```suggestion
| Polygon.Analysis.Name | File name. | string |
| Polygon.Analysis.Size | File size. | number |
```

File: Packs/Polygon/ReleaseNotes/1_0_11.md [hunk 14]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+
+##### Detonate File - Group-IB TDS Polygon
+
+- Added a new task for filtering file types supported by Polygon.
```
Comment (ShirleyDenkberg) [inline Packs/Polygon/ReleaseNotes/1_0_11.md:6]:
```suggestion
Added a new task for filtering file types supported by Polygon.
```

**Comment by ShirleyDenkberg** [comment]
@ssokolovich Doc review completed.

**Pull request merged**